### PR TITLE
🔧 sbref generation update and desc-preproc/sbref BOLD resource pool naming changes

### LINKED ISSUE #1772: 🐛CPAC should be able to preprocess raw BIDS images that are in ARI format
### Describe the bug

When the raw input data is in ARI format, the motion reference data takes the ARI orientation but the functional motion data will be in the RPI orientation. Because of this discrepancy in the orientation of the functional image and the reference image, the pipeline will crash during 3dvolreg.

This happens because when `calculate motion first` is set, the `func_preproc `collection of blocks, including `func_reorient`, happens after the motion ref creation. (Refer [here](https://github.com/FCP-INDI/C-PAC/blob/12c4f74a3cfeb544239e34301853b90b34d7edc1/CPAC/pipeline/cpac_pipeline.py#L1175-L1176)), hence the  reference image takes the orientation of the raw BIDS data. 

We can account for this by adding `func_reorient` to [func_init_blocks](https://github.com/FCP-INDI/C-PAC/blob/12c4f74a3cfeb544239e34301853b90b34d7edc1/CPAC/pipeline/cpac_pipeline.py#L1126-L1129). By doing so functional reorientation will occur before motion reference creation and both the functional preprocessed and the bold reference image will be in the same orientation. 

### To reproduce

To reproduce this error. 
1) Use Raw BIDS data that are in ARI orientation as input. 
2) Run CPAC on docker with the images in raw BIDS images in ARI as input. 

### Preconfig

- [ ] default
- [ ] abcd-options
- [ ] anat-only
- [ ] blank
- [ ] ccs-options
- [ ] fmriprep-options
- [ ] fx-options
- [ ] monkey
- [ ] monkey-ABCD
- [ ] ndmg
- [ ] nhp-macaque
- [ ] preproc
- [ ] rbc-options
- [ ] rodent

### Custom pipeline configuration

```
%YAML 1.1
---
# CPAC Pipeline Configuration YAML file
# Version 1.8.4.dev
#
# http://fcp-indi.github.io for more info.
#
# Tip: This file can be edited manually with a text editor for quick modifications.

FROM: default


pipeline_setup: 
  # Name for this pipeline configuration - useful for identification.
  pipeline_name: cpac_preproc

  working_directory: 
    # Deletes the contents of the Working Directory after running.
    # This saves disk space, but any additional preprocessing or analysis will have to be completely re-run.
    remove_working_dir: Off

  system_config: 
    # Select Off if you intend to run CPAC on a single machine.
    # If set to On, CPAC will attempt to submit jobs through the job scheduler / resource manager selected below.
    on_grid: 
      SGE: 
        # SGE Parallel Environment to use when running CPAC.
        # Only applies when you are running on a grid or compute cluster using SGE.
        parallel_environment: cpac

    # The maximum amount of memory each participant's workflow can allocate.
    # Use this to place an upper bound of memory usage.
    # - Warning: 'Memory Per Participant' multiplied by 'Number of Participants to Run Simultaneously'
    #   must not be more than the total amount of RAM.
    # - Conversely, using too little RAM can impede the speed of a pipeline run.
    # - It is recommended that you set this to a value that when multiplied by
    #   'Number of Participants to Run Simultaneously' is as much RAM you can safely allocate.
    maximum_memory_per_participant: 3

    # Full path to the FSL version to be used by CPAC.
    # If you have specified an FSL path in your .bashrc file, this path will be set automatically.
    FSLDIR: FSLDIR

  Amazon-AWS: 
    # Enable server-side 256-AES encryption on data to the S3 bucket
    s3_encryption: On

anatomical_preproc: 
  # Non-local means filtering via ANTs DenoiseImage
  non_local_means_filtering:
    run: Off

  # N4 bias field correction via ANTs
  n4_bias_field_correction:
    run: On

segmentation: 
  tissue_segmentation: 
    # option parameters
    FSL-FAST: 
      use_priors: 
        # Full path to a directory containing binarized prior probability maps.
        # These maps are included as part of the 'Image Resource Files' package available on the Install page of the User Guide.
        # It is not necessary to change this path unless you intend to use non-standard priors.
        priors_path: $FSLDIR/data/standard/tissuepriors/2mm

    Template_Based: 
      # These masks should be in the same space of your registration template, e.g. if
      # you choose 'EPI Template' , below tissue masks should also be EPI template tissue masks.
      #
      # Options: ['T1_Template', 'EPI_Template']
      template_for_segmentation: []

      # These masks are included as part of the 'Image Resource Files' package available
      # on the Install page of the User Guide.

      # Full path to a binarized White Matter mask.
      WHITE: $FSLDIR/data/standard/tissuepriors/2mm/avg152T1_white_bin.nii.gz

      # Full path to a binarized Gray Matter mask.
      GRAY: $FSLDIR/data/standard/tissuepriors/2mm/avg152T1_gray_bin.nii.gz

      # Full path to a binarized CSF mask.
      CSF: $FSLDIR/data/standard/tissuepriors/2mm/avg152T1_csf_bin.nii.gz

registration_workflows: 
  functional_registration: 
    EPI_registration: 
      ANTs: 
        # EPI registration configuration - synonymous with T1_registration
        # parameters under anatomical registration above
        parameters:

functional_preproc: 
  truncation: 
    # First timepoint to include in analysis.
    # Default is 0 (beginning of timeseries).
    # First timepoint selection in the scan parameters in the data configuration file, if present, will over-ride this selection.
    # Note: the selection here applies to all scans of all participants.
    start_tr: 3

    # Last timepoint to include in analysis.
    # Default is None or End (end of timeseries).
    # Last timepoint selection in the scan parameters in the data configuration file, if present, will over-ride this selection.
    # Note: the selection here applies to all scans of all participants.
    stop_tr: End

  motion_estimates_and_correction: 
    motion_estimates: 
      # calculate motion statistics BEFORE slice-timing correction
      calculate_motion_first: On

  func_masking: 
    # using: ['AFNI', 'FSL', 'FSL_AFNI', 'Anatomical_Refined', 'Anatomical_Based']
    # this is a fork point
    using: [Anatomical_Refined]

 
nuisance_corrections: 
  2-nuisance_regression: 
    # Select which nuisance signal corrections to apply
    Regressors: 
      - Bandpass:
          bottom_frequency: 0.01
          top_frequency: 0.1
        CerebrospinalFluid:
          erode_mask: On
          extraction_resolution: 2
          summary: Mean
        Motion:
          include_delayed: On
          include_delayed_squared: On
          include_squared: On
        Name: Regressor-2
        PolyOrt:
          degree: 2
        aCompCor:
          extraction_resolution: 2
          summary:
            components: 5
            method: DetrendPC
          tissues:
          - WhiteMatter
          - CerebrospinalFluid

    # Standard Lateral Ventricles Binary Mask
    # used in CSF mask refinement for CSF signal-related regressions
    lateral_ventricles_mask: $FSLDIR/data/atlases/HarvardOxford/HarvardOxford-lateral-ventricles-thr25-2mm.nii.gz

timeseries_extraction: 
  run: Off

seed_based_correlation_analysis: 
  # SCA - Seed-Based Correlation Analysis
  # For each extracted ROI Average time series, CPAC will generate a whole-brain correlation map.
  # It should be noted that for a given seed/ROI, SCA maps for ROI Average time series will be the same.
  run: Off

amplitude_low_frequency_fluctuation: 
  # ALFF & f/ALFF
  # Calculate Amplitude of Low Frequency Fluctuations (ALFF) and and fractional ALFF (f/ALFF) for all voxels.
  run: Off

regional_homogeneity: 
  # ReHo
  # Calculate Regional Homogeneity (ReHo) for all voxels.
  run: On

voxel_mirrored_homotopic_connectivity: 
  # VMHC
  # Calculate Voxel-mirrored Homotopic Connectivity (VMHC) for all voxels.
  run: Off

  symmetric_registration: 
    # Included as part of the 'Image Resource Files' package available on the Install page of the User Guide.
    # It is not necessary to change this path unless you intend to use a non-standard symmetric template.
    T1w_brain_template_symmetric: $FSLDIR/data/standard/MNI152_T1_${resolution_for_anat}_brain_symmetric.nii.gz

    # A reference symmetric brain template for resampling
    T1w_brain_template_symmetric_for_resample: $FSLDIR/data/standard/MNI152_T1_1mm_brain_symmetric.nii.gz

    # Included as part of the 'Image Resource Files' package available on the Install page of the User Guide.
    # It is not necessary to change this path unless you intend to use a non-standard symmetric template.
    T1w_template_symmetric: $FSLDIR/data/standard/MNI152_T1_${resolution_for_anat}_symmetric.nii.gz

    # A reference symmetric skull template for resampling
    T1w_template_symmetric_for_resample: $FSLDIR/data/standard/MNI152_T1_1mm_symmetric.nii.gz

    # Included as part of the 'Image Resource Files' package available on the Install page of the User Guide.
    # It is not necessary to change this path unless you intend to use a non-standard symmetric template.
    dilated_symmetric_brain_mask: $FSLDIR/data/standard/MNI152_T1_${resolution_for_anat}_brain_mask_symmetric_dil.nii.gz

    # A reference symmetric brain mask template for resampling
    dilated_symmetric_brain_mask_for_resample: $FSLDIR/data/standard/MNI152_T1_1mm_brain_mask_symmetric_dil.nii.gz

network_centrality: 
  # Calculate Degree, Eigenvector Centrality, or Functional Connectivity Density.
  run: Off

# PACKAGE INTEGRATIONS
# --------------------
PyPEER: 
  # Template-space eye mask
  eye_mask_path: $FSLDIR/data/standard/MNI152_T1_${func_resolution}_eye_mask.nii.gz

```

### Run command

docker run --user 1136:1054 --security-opt=apparmor:unconfined -i --rm -v /data3/autism_center/wave1/QC/cpac_1.8.3:/scratch -v /data3/autism_center/wave1/QC/cpac_1.8.3:/outputs -v /data3/autism_center/wave1/QC/cpac_1.8.3/config:/config -v /data3/autism_center/wave1/BIDS:/bids_dataset fcpindi/c-pac:release-v1.8.3 /bids_dataset /outputs participant --pipeline_file /config/pipeline_config_cpac_1.8.3.yml --save_working_dir --save_working_dir --tracking_opt-out --n_cpus 25 --mem_gb 20 --participant_label 30002

### Expected behavior

The pipeline will not run to completion and you will get an error message and a crash file will be generated. 

You will get the following error: 
```
Error of subject workflow cpac_sub-30002_ses-awake
CPAC run error:
  Pipeline configuration: cpac_preproc
  Subject workflow: cpac_sub-30002_ses-awake
  Elapsed run time (minutes): 7.66220289071401
  Timing information saved in /outputs/log/cpac_individual_timing_cpac_preproc.csv
  System time of start:   2022-09-02 15:21:58
Traceback (most recent call last):
 File “/code/CPAC/pipeline/cpac_pipeline.py”, line 515, in run_workflow
  workflow.run(plugin=plugin, plugin_args=plugin_args)
 File “/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/engine/workflows.py”, line 632, in run
  runner.run(execgraph, updatehash=updatehash, config=self.config)
 File “/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/plugins/base.py”, line 192, in run
  report_nodes_not_run(notrun)
 File “/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/plugins/tools.py”, line 98, in report_nodes_not_run
  (“Workflow did not execute cleanly. ” “Check log for details”)
RuntimeError: Workflow did not execute cleanly. Check log for details
/code/CPAC/utils/monitoring/draw_gantt_chart.py:64: Warning: ‘Event logged twice or events started at exact same time!’
 warn(str(KeyError(err_msg)), category=Warning)
/code/CPAC/nuisance/bandpass.py:82: ComplexWarning: Casting complex values to real discards the imaginary part
 Y_bp[:, j] = ideal_bandpass(Yc[:, j], sample_period, bandpass_freqs)
/code/CPAC/nuisance/bandpass.py:82: ComplexWarning: Casting complex values to real discards the imaginary part
 Y_bp[:, j] = ideal_bandpass(Yc[:, j], sample_period, bandpass_freqs)
```
And a crash file with the following information is generated:
```
Standard error:
++ 3dvolreg: AFNI version=AFNI_22.0.09 (Feb 10 2022) [64-bit]
++ Authored by: RW Cox
*+ WARNING:   If you are performing spatial transformations on an oblique dset,
  such as /outputs/working/cpac_sub-30002_ses-awake/_scan_rest_run-04/func_get_mean_RPI_64/sub-30002_ses-awake_task-rest_run-04_bold_calc_tstat.nii.gz,
  or viewing/combining it with volumes of differing obliquity,
  you should consider running: 
     3dWarp -deoblique 
  on this and  other oblique datasets in the same session.
 See 3dWarp -help for details.
++ Oblique dataset:/outputs/working/cpac_sub-30002_ses-awake/_scan_rest_run-04/func_get_mean_RPI_64/sub-30002_ses-awake_task-rest_run-04_bold_calc_tstat.nii.gz is 6.999949 degrees from plumb.
** Input /outputs/working/cpac_sub-30002_ses-awake/_scan_rest_run-04/func_motion_correct_3dvolreg_80/mapflow/_func_motion_correct_3dvolreg_8016/sub-30002_ses-awake_task-rest_run-04_bold_calc_tshift_resample_16.nii.gz and base /outputs/working/cpac_sub-30002_ses-awake/_scan_rest_run-04/func_get_mean_RPI_64/sub-30002_ses-awake_task-rest_run-04_bold_calc_tstat.nii.gz don't have same dimensions!
   Input: nx=64  ny=80  nz=34
   Base:  nx=80  ny=64  nz=34

** Input /outputs/working/cpac_sub-30002_ses-awake/_scan_rest_run-04/func_motion_correct_3dvolreg_80/mapflow/_func_motion_correct_3dvolreg_8016/sub-30002_ses-awake_task-rest_run-04_bold_calc_tshift_resample_16.nii.gz and base /outputs/working/cpac_sub-30002_ses-awake/_scan_rest_run-04/func_get_mean_RPI_64/sub-30002_ses-awake_task-rest_run-04_bold_calc_tstat.nii.gz don't have same grid spacing!
   Input: dx= 3.000  dy=-3.000  dz= 3.000
   Base:  dx= 3.000  dy= 3.000  dz= 3.000
** Input /outputs/working/cpac_sub-30002_ses-awake/_scan_rest_run-04/func_motion_correct_3dvolreg_80/mapflow/_func_motion_correct_3dvolreg_8016/sub-30002_ses-awake_task-rest_run-04_bold_calc_tshift_resample_16.nii.gz and base /outputs/working/cpac_sub-30002_ses-awake/_scan_rest_run-04/func_get_mean_RPI_64/sub-30002_ses-awake_task-rest_run-04_bold_calc_tstat.nii.gz don't have same orientation!
   Input: R-L P-A I-S
   Base:  A-P R-L I-S 
** FATAL ERROR: perhaps you could make your datasets match?
```

### Acceptance criteria

The user should be able to preprocess raw BIDS images that are in ARI orientation. 
1. User provides Raw functional and anatomical images that are in ARI orientation.
2. User uses the default CPAC config to preprocess the images and the pipeline runs to completion without any errors.

### Screenshots

_No response_

### C-PAC version

_No response_

### Container platform

_No response_

### Docker and/or Singularity version(s)

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by sgiavasis**
## Fixes

Fixes #1772 by @tergeorge, #1780 by @nathaliaesper, and #1783 by @nimzodisaster
Related to #1242 by @sgiavasis 

## Description
- Removes all extraneous preprocessed BOLD namings in the node block docstring input parameters, moving everything to the leaf resource `desc-preproc_bold`
- Moves over to using `sbref` as the naming for bold registration input
- Brings together the disparate options for bold registration inputs under `sbref`
- Updates distortion correction options to deal directly with `sbref`
- Move `func_reorient` back to its rightful place
- Generates `sbref` (via the `coregistration_prep_{etc.}` blocks) before distortion correction

## Technical details

- Going forward, will need to rely on JSON meta-data with `SkullStripped: True/False` regarding skullstrip status, now that `desc-brain` has been removed

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the `develop` branch of the repository. 
- [x] My commit messages follow best practices.
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I updated the changelog.
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by shnizzedy — APPROVED** [review]
Looks good to me, and the failing tests are unrelated, fixed in #1803

**Comment by sgiavasis** [comment]
> Looks good to me, and the failing tests are unrelated, fixed in #1803

@shnizzedy Thanks! Though I just pushed some final changes - need one last re-review.

File: CPAC/registration/registration.py [hunk 1]
```diff
@@ -2806,12 +2805,13 @@ def coregistration(wf, cfg, strat_pool, pipe_num, opt=None):
...
         'functional_registration']['coregistration']['interpolation']
 
-        node, out = strat_pool.get_data('desc-reginput_bold')
+        node, out = strat_pool.get_data('sbref')
         wf.connect(node, out, func_to_anat, 'inputspec.func')
 
         if cfg.registration_workflows['functional_registration'][
             'coregistration']['reference'] == 'brain':
-            node, out = strat_pool.get_data('desc-brain_T1w')
+            # TODO: use JSON meta-data to confirm
```
Comment (shnizzedy) [inline CPAC/registration/registration.py:2813]: We should probably open an issue to try make sure this doesn't get lost in the shuffle

**Review by shnizzedy — APPROVED** [review]
This is really nice. I think the provenance will be easier to read after this too!

File: CPAC/registration/registration.py (hunk 1 above)
Comment (sgiavasis) [reply CPAC/registration/registration.py:2813]: Agreed. Though I won't be forgetting personally, as it's been stinging my mind for awhile now as part of my engine redesign finalization 😄

**Pull request merged**